cardano-cli/src/Cardano/CLI/Types/Errors/NodeEraMismatchError.hs

```diff
@@ -0,0 +1,15 @@
+{-# LANGUAGE GADTs #-}
+
+module Cardano.CLI.Types.Errors.NodeEraMismatchError
+  ( NodeEraMismatchError(..)
+  ) where
+
+import           Cardano.Api
+
+data NodeEraMismatchError = forall era nodeEra value.
+  Show value =>
+    NodeEraMismatchError
+    { originalValue :: !value
```

Comment:
I think we don't need the `originalValue` field. It's probably enough to say the node is in an unexpected era and what era that is.

Reply:
The field is so that if we have some value that is in the wrong era for example `TxOut era` then we can print out a useful error message to say what it was that was in the wrong era.  @Jimbo4350 is right that we don't need it because we now outright reject circumstances where eras don't match rather than before when we were more lenient and additional context was helpful.